__tests__/app.test.js

```diff
@@ -0,0 +1,111 @@
+    })
+    test('should return the topics from the table in nc_news_test db with both properties, slug and description', ()=>{
+        return request(app)
+        .get('/api/topics')
+        .then(({body})=>{
+            const topicsArr = body.topics
+            const exampleTopic = {
+                description: 'what books are made of',
+                slug: 'paper'
+              }
+            topicsArr.forEach((topic)=> {
+                for (const key of Object.keys((exampleTopic))){
+                    expect(Object.hasOwn(topic, key))
+                }
+            })
+        })
+    })
+    
+
+})
+describe('GET /api', ()=>{
+    test('should return an object with all valid endpoints and their methods as properties',()=>{
+        return request(app)
+        .get('/api')
+        .then(({body})=>{
+            const appRouteNames = []
+            app._router.stack.forEach((r)=>{
+                if(r.route && r.route.path){
+                    if(r.route.path !== '/*'){ //exclude invalid endpoints
+                        Object.keys(r.route.methods)
+                        .forEach((method) => {
+                            appRouteNames.push(`${method.toUpperCase()} ${r.route.path}`)
+                        })
+                    }
+                } 
+            })
+            let hasAllProperties = true
+            appRouteNames.forEach((route)=>{
+                if (!Object.keys(endpoints).includes(route)) hasAllProperties = false
+            })
+            expect(hasAllProperties).toBe(true)
+        })
+    })
+})
+
+describe('GET /api/articles/:article_id', ()=>{
+    test('sould return an article given the correct id or an empty array if no such id exists', ()=>{
+        return request(app)
+        .get('/api/articles/3').then(({body})=>{
+            expect.objectContaining({
+                article_id: expect.any(Number),
+                title: expect.any(String),
+                topic: expect.any(String),
+                author: expect.any(String),
+                body: expect.any(String),
+                created_at: expect.any(String),
+                votes: expect.any(Number),
+                article_img_url: expect.any(String)
+              })
+        })
+    })
+    test('sould return an error status 400 with appropriate message if the id is invalid', ()=>{
+        return request(app)
+        .get('/api/articles/invalid_id')
+        .expect(400)
+        .then(({body})=>{
+           expect(body).toEqual({ message: 'Invalid article_id - the article does not exist or your article_id is malformed' }) 
+        })
+    })
+
+    test('should return an error status 400 with appropriate message if the id does not exist', ()=>{
+        return request(app)
+        .get('/api/articles/9999')
+        .expect(400)
+        .then(({body})=>{
+            expect(body).toEqual({ message: 'Invalid article_id - the article does not exist or your article_id is malformed' })
+        })
+    })
```

Comment:
This test should be a 404 - the request itself is valid - it is for a number, but the article doesn't exist. You can reject this with a 404 status in your model if there are no articles sent back.